### PR TITLE
4 add Jenkins setup

### LINKED ISSUE #3: Remove Edge First-Run in Nexus script
We are using the new Edge (Chromium) in this setup, as it is a Microsoft product, and thus preferred by our users when setting up Windows Servers.

As such, it would be nice to _not_ have Edge prompt for setup on first-run. This will reduce the number of clicks. Also, these are servers, so the Admins will likely NOT be logging to their Microsoft accounts on the browser.

### LINKED ISSUE #4: Add Jenkins setup
We still require the addition of the Jenkins setup to this server. This should include:

- Choco install of the package
- Version-pinning, if required
- Pre-downloaded Jenkins scripts for Package Internalizer automation
- Pre-defined Jenkins jobs for the scripts above

### LINKED ISSUE #7: Add ChocolateyTest NuGet repo
Current environments (like QDE) have a test repo setup as a best practice.

We should add one here to keep things consistent. This will benefit Jenkins setup staying the same as other env's as well.

### LINKED ISSUE #8: Add ability to pass JSON data between scripts
Ideally, there is data we'd like to pass between scripts (e.g. NuGet API key), that would be helpful for the user and Support to have handy. We already have transcript files, but those aren't always easy to get this info from.

For the time being, we can output these as JSON settings files from each script, that other scripts can read from.

We can always iterate a different direction later.

### LINKED ISSUE #10: Update Start-C4bSetup.ps1 to not use short URI's
Currently, the `Start-C4bSetup.ps1` script utilizes `ch0.co` short URI's as a convenience for the user.  As some organizations block the usage of short URI's for security (although one could argue that's counter-intuitive), it would behoove us to use the raw file URI's directly from this repo.

### PR DESCRIPTION
**Pull request by adilio**
This PR addresses the following:

- Closes #4 - Add Jenkins Setup
    - Choco install of the package, pinned to version
    - Pre-downloaded Jenkins scripts for Package Internalizer automation
    - Pre-defined Jenkins jobs for the scripts above
    - Output data to JSON to pass between scripts

As a result of necessary changes for the above, this PR also does the following:

- Closes #3 - Remove Edge First-Run in Nexus script
- Closes #7 - Add ChocolateyTest NuGet repo
- Closes #8 - Add ability to pass JSON data between scripts
- Closes #10 - Update Start-C4bSetup.ps1 to not use short URI's

### CODE REVIEW

File: Start-C4bJenkinsSetup.ps1 [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+foreach ($PluginName in $JenkinsPlugins.Keys) {
+    $PluginUri = 'https://updates.jenkins-ci.org/download/plugins/{0}/{1}/{0}.hpi' -f $PluginName, $JenkinsPlugins[$PluginName]
+    $PluginPath = '{0}/plugins/{1}.hpi' -f $jenkinsHome, $PluginName
+
+    Invoke-WebRequest -Uri $PluginUri -OutFile $PluginPath -UseBasicParsing
+}
+
+# Restore default progress bar setting
+$ProgressPreference = 'Continue'
```
Comment (steviecoaster) [inline Start-C4bJenkinsSetup.ps1:145]: We could avoid IWR entirely here, and be faster
```suggestion
foreach($PluginName in $JenkinsPlugins.Keys) {
  $PluginUri = 'https://updates.jenkins-ci.org/download/plugins/{0}/{1}/{0}.hpi' -f $PluginName,$JenkinsPlugins[$PluginName]
  $PluginPath = '{0}/plugins/{1}.hpi -f $JenkinsHome, $PluginName
  
  [System.Net.WebClient]::New().DownloadFile($PluginUri,$PluginPath)
```

File: Start-C4bCcmSetup.ps1 [hunk 2]
```diff
@@ -172,21 +172,21 @@ choco pin add --name="'dotnetcore-windowshosting'" --version="'2.2.7'" --reason=
...
+    CCMWebPortal = "http://localhost/Account/Login"
+    DefaultUser = "ccmadmin"
+    DefaultPwToBeChanged = "123qwe"
+    CCMDBUser = $DatabaseUser
 }
+$CcmJson | ConvertTo-Json | Out-File .\ccm.json
 
-#Stop logging
-Stop-Transcript
+# Completion notice
```
Comment (steviecoaster) [inline Start-C4bCcmSetup.ps1:185]: This comment seems unnecessary

**Review by steviecoaster — CHANGES_REQUESTED** [review]

File: Start-C4bJenkinsSetup.ps1 (hunk 1 above)
Comment (adilio) [reply Start-C4bJenkinsSetup.ps1:145]: This is a good idea. Two things:
- This is straight from [code](https://github.com/chocolatey/qde-azure/blob/main/application/source/Install-Jenkins.ps1) by @JPRuskin and @Windos, so just tagging them for awareness
- We can probably get rid of the Progress preference then too

File: Start-C4bCcmSetup.ps1 (hunk 2 above)
Comment (adilio) [reply Start-C4bCcmSetup.ps1:185]: Good call. We should probably remove these in all the scripts then.

**Comment by ryanrichter94** [comment]
LGTM 🚀

**Pull request merged**